pkg/dao/rpms.go

```diff
@@ -162,35 +162,23 @@ func (r rpmDaoImpl) Search(orgID string, request api.ContentUnitSearchRequest) (
 	urls := handleTailChars(request)
 	uuids := request.UUIDs
 
-	// This implement the following SELECT statement:
-	//
-	// SELECT DISTINCT ON (rpms.name)
-	//        rpms.name, rpms.summary
-	// FROM rpms
-	//      inner join repositories_rpms on repositories_rpms.rpm_uuid = rpms.uuid
-	//      inner join repositories on repositories.uuid = repositories_rpms.repository_uuid
-	//      left join repository_configurations on repository_configurations.repository_uuid = repositories.uuid
-	// WHERE (repository_configurations.org_id = 'acme' OR repositories.public)
-	//       AND ( repositories.url in (...)
-	//             OR repository_configurations.uuid in (...)
-	//       )
-	//       AND rpms.name LIKE 'demo%'
-	// ORDER BY rpms.name, rpms.epoch DESC
-	// LIMIT 20;
+	// Lookup repo uuids to search
+	repoUuids := []string{}
+	orGroupPublicPrivatePopular := r.db.Where("repository_configurations.org_id = ?", orgID).Or("repositories.public").Or("repositories.url in ?", popularRepoUrls())
+	r.db.Model(&models.Repository{}).
+		Joins("left join repository_configurations on repositories.uuid = repository_configurations.repository_uuid and repository_configurations.org_id = ?", orgID).
+		Where(orGroupPublicPrivatePopular).
```

Comment:
Note that I think this "and repository_configurations.org_id = ?" is actually what is causing most of the speed up here.  Previously i think it was causing the join to include EVERY popular repo_config across orgs in the join now that the orGroupPublicPrivatePopular includes all popular repo.  

I split out the query to lookup the repo UUIDs because i thought it simplified the query quite a bit and made it easier to understand